src/tree-controller.ts

```diff
@@ -0,0 +1,52 @@
+import { TreeService } from './tree.service';
+import { Tree } from './tree';
+import { TreeModel } from './tree.types';
+import { NodeMenuItemAction } from './menu/menu.events';
+import { NodeEditableEvent, NodeEditableEventAction } from './editable/editable.events';
+import { TreeInternalComponent } from './tree-internal.component';
+import { Subject, Observable } from 'rxjs/Rx';
+import { Injectable, Inject, ElementRef } from '@angular/core';
+import * as _ from 'lodash';
+
+export class TreeController {
+  private tree: Tree;
+
+  constructor (private treeInternalComponent: TreeInternalComponent) {
+    this.tree = this.treeInternalComponent.tree;
+  }
+
+  public select(e: MouseEvent = new MouseEvent('click')): void {
+    if (this.treeInternalComponent && !this.treeInternalComponent.isSelected && typeof this.treeInternalComponent.onNodeSelected === 'function') {
+      this.treeInternalComponent.onNodeSelected(e);
+    }
+  }
+
+  public expand(): void {
+    if (this.treeInternalComponent && !this.tree.isNodeExpanded()) {
```

Comment:
@lovery you don't need to check `this.treeInternalComponent` cause if it doesn't exist 
this line in constructor `this.tree = this.treeInternalComponent.tree;` simple won't allow to create an instance of the TreeController